boss-ws/src/main/java/org/genomebridge/boss/http/BossApplication.java

```diff
@@ -66,21 +66,37 @@ public void run(BossConfiguration config, Environment env) throws Exception {
         // Create an API object that the resources can use.
         gDBI = new DBIFactory().build(env, config.getDataSourceFactory(), "db");
         gDBI.registerArgumentFactory(new NullArgumentFactory());
-        ObjectStoreConfiguration localConf = config.getLocalStoreConfiguration();
-        ObjectStore localStore = getObjectStore(localConf);
-        ObjectStoreConfiguration cloudConf = config.getCloudStoreConfiguration();
-        ObjectStore cloudStore = getObjectStore(cloudConf);
-        gBossAPI = new DatabaseBossAPI(gDBI,localStore,cloudStore,getMessages());
+        Map<String,ObjectStoreConfiguration> objectStoreConfigurationMap = config.getObjectStores();
+        gObjectStores = getObjectStoresMap(objectStoreConfigurationMap);
+        gBossAPI = new DatabaseBossAPI(gDBI,gObjectStores,getMessages());
         SwaggerConfiguration swagger = config.getSwaggerConfiguration();
         // Set up the resources themselves.
         env.jersey().register(new ObjectResource(gBossAPI));
         env.jersey().register(new AllObjectsResource(gBossAPI));
         setSwaggerConfiguration(config, env, swagger);
-        if ( localConf.type == ObjectStoreType.FCS || cloudConf.type == ObjectStoreType.FCS ) {
-            env.jersey().register(new FCSResource());
-        }
+
+        for (Map.Entry<String, ObjectStoreConfiguration> entry : objectStoreConfigurationMap.entrySet()) {
```

Comment:
In the new code, it's similar.  We check if there exists some object store with type FCS, and if it exists, we register an FCSResource, I'm noticed that we never break the loop. In order to register one, and only one resource, we should add a break clause.